src/runtime/rpc/rpc_message.cpp

```diff
@@ -252,6 +252,7 @@ message_ex *message_ex::copy(bool clone_content, bool copy_for_receive)
 
         if ((const char *)header != buffers[0].data()) {
             memcpy(ptr, (const void *)header, sizeof(message_header));
+            i += sizeof(message_header);
```

Comment:
How about removing the `i` variable totally, you can compare the offset of `ptr` and `recv_buffer` instead in line 264.

Reply:
I have made the changes, could you please help me take a look again.

Reply:
Thanks for the contribution!